File: VB/SecuritySystemExample.Module/Module.vb
Imports System
Imports System.Collections.Generic

Imports DevExpress.ExpressApp
Imports System.Reflection
Imports DevExpress.ExpressApp.Security.Strategy
Imports DevExpress.Persistent.BaseImpl.PermissionPolicy

Namespace PermissionPolicyExample.Module
    Public NotInheritable Partial Class PermissionPolicyExampleModule
        Inherits ModuleBase

        Public Sub New()
            InitializeComponent()
        End Sub
        Protected Overrides Function GetDeclaredExportedTypes() As IEnumerable(Of Type)
            Dim result As New List(Of Type)(MyBase.GetDeclaredExportedTypes())
            result.AddRange(New Type() { GetType(PermissionPolicyUser), GetType(PermissionPolicyRole) })
            Return result
        End Function
    End Class
End Namespace

File: CS/SecuritySystemExample.Module/Module.cs
using System;
using System.Collections.Generic;

using DevExpress.ExpressApp;
using System.Reflection;
using DevExpress.ExpressApp.Security.Strategy;


namespace SecuritySystemExample.Module {
    public sealed partial class SecuritySystemExampleModule : ModuleBase {
        public SecuritySystemExampleModule() {
            InitializeComponent();
        }
        protected override IEnumerable<Type> GetDeclaredExportedTypes() {
            List<Type> result = new List<Type>(base.GetDeclaredExportedTypes());
            result.AddRange(new Type[] { typeof(SecuritySystemUser), typeof(SecuritySystemRole) });
            return result;
        }
    }
}
